vboxtouch/vboxtouch.cpp

```diff
@@ -112,8 +114,83 @@ const static vbox_set_mouse_status set_mouse_status = {
     sizeof(vbox_header), 0, { VBOXMOUSE_WANT_ABSOLUTE | VBOXMOUSE_NEW_PROTOCOL | VBOXMOUSE_HOST_DRAWS_CURSOR }
 };
 
+QRect screenGeometryFromFramebuffer()
+{
+    const QSize uiSize(QGuiApplication::primaryScreen()->size());
+    QRect defaultGeometry(QPoint(), uiSize);
+
+    /* Read framebuffer dimentions from device
+     * in case of VM resolution in VM extra data (CustomVideoMode1, GUI/LastGuestSizeHint)
+     * is different than resolution reported by QGuiApplication::primaryScreen()->geometry().size()
+     */
+    const QString fbDevice = "/dev/fb0";
+    const int fh = open(fbDevice.toLocal8Bit().constData(), O_RDONLY);
+    if (fh < 0) {
+        qWarning("vboxtouch: cannot open framebuffer %s: %s", qPrintable(fbDevice), strerror(errno));
+        return defaultGeometry;
+    }
+
+    uint32_t var[40]; // NOTE: Not a real struct
```

Comment:
Why don't you just include linux/fb.h and use the real thing? You also wouldn't have to define FBIOGET_VSCREENINFO yourself

Reply:
Done.